src/server/scripts/Northrend/Gundrak/boss_moorabi.cpp

```diff
@@ -125,41 +141,50 @@ class boss_moorabi : public CreatureScript
-                        if (_transformed)
-                            DoCastAOE(SPELL_QUAKE);
-                        else
-                            DoCastAOE(SPELL_GROUND_TREMOR);
-                        events.ScheduleEvent(eventId, 10 * IN_MILLISECONDS);
-                        break;
-                    case EVENT_NUMBLING_SHOUT:
-                        if (_transformed)
-                            DoCastAOE(SPELL_NUMBING_ROAR);
-                        else
-                            DoCastAOE(SPELL_NUMBING_SHOUT);
-                        events.ScheduleEvent(eventId, 10 * IN_MILLISECONDS);
-                        break;
-                    case EVENT_DETERMINED_STAB:
-                        if (_transformed)
-                            DoCastVictim(SPELL_DETERMINED_GORE);
-                        else
-                            DoCastVictim(SPELL_DETERMINED_STAB);
-                        events.ScheduleEvent(eventId, 8 * IN_MILLISECONDS);
-                        break;
-                    case EVENT_TRANFORMATION:
-                        Talk(EMOTE_BEGIN_TRANSFORM);
-                        Talk(SAY_TRANSFORM);
-                        DoCast(me, SPELL_TRANSFORMATION);
-                        events.ScheduleEvent(eventId, 10 * IN_MILLISECONDS);
-                        break;
-                    default:
-                        break;
+                    switch (eventId)
+                    {
+                        case EVENT_GROUND_TREMOR:
+                            if (roll_chance_i(50))
+                                Talk(SAY_QUAKE);
+                            DoCastAOE(_transformed ? SPELL_QUAKE : SPELL_GROUND_TREMOR);
+                            events.Repeat(Seconds(10));
+                            break;
+                        case EVENT_NUMBLING_SHOUT:
+                            DoCastAOE(_transformed ? SPELL_NUMBING_ROAR : SPELL_NUMBING_SHOUT);
+                            events.Repeat(Seconds(10));
+                            break;
+                        case EVENT_DETERMINED_STAB:
+                            DoCastAOE(_transformed ? SPELL_DETERMINED_GORE : SPELL_DETERMINED_STAB);
+                            events.Repeat(Seconds(8));
+                            break;
+                        case EVENT_TRANFORMATION:
+                            Talk(EMOTE_BEGIN_TRANSFORM);
+                            Talk(SAY_TRANSFORM);
+                            DoCastSelf(SPELL_TRANSFORMATION);
+                            DoCastSelf(SPELL_SUMMON_PHANTOM_TRANSFORM, true);
+                            events.Repeat(Seconds(10));
+                            break;
+                        case EVENT_PHANTOM:
+                            DoCastSelf(SPELL_SUMMON_PHANTOM, true);
+                            events.Repeat(Seconds(20), Seconds(25));
+                            break;
+                        default:
+                            break;
+                    }
```

Comment:
`events.Update(diff);`
`if (me->HasUnitState(UNIT_STATE_CASTING))`
` return;`
Is this one still needed then?